src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs

```diff
@@ -501,7 +501,7 @@ bool IEqualityComparer<string>.Equals(string? extensionA, string? extensionB)
                 //with the rules for comparing/normalizing partnames.
                 //Refer to PackUriHelper.ValidatedPartUri.GetNormalizedPartUri method.
                 //Currently normalization just involves upper-casing ASCII and hence the simplification.
-                return (string.CompareOrdinal(extensionA.ToUpperInvariant(), extensionB.ToUpperInvariant()) == 0);
+                return extensionA.ToUpperInvariant() == extensionB.ToUpperInvariant();
```

Comment:
nit: could be equals ignore case since the comment mentions ASCII?

Reply:
Yeah, but there's that scary comment about keeping all logic here in sync, so... yeah... not going to risk anything here. :)